### ISSUE TITLE
`$params` are not cleared when adding, removing collection items

### ISSUE THREAD

**Issue by martinszeltins** [victorgarciaesgi/regle#21]
I noticed that as I validate my form, the `$params` keep growing with more entries but are never cleared. I attached a screencast where you can see that as I validate, the `$params` keep growing but are never cleared, even when I remove all items from the collection.

**Reproduction:**
https://stackblitz.com/~/github.com/martinszeltins/regle-test-11

**Screencast:**

https://github.com/user-attachments/assets/f15fdaa2-d402-47ff-a0b4-d573b0bfe8a7

````
"$rules": {
  "required": {
    "$active": true,
    "$message": "This field is required",
    "$type": "required",
    "$params": [
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false
  ],
````

**Comment by victorgarciaesgi** [comment]
That's a good catch!! Thanks i'll look

**Comment by martinszeltins** [comment]
It's getting harder for me to find new bugs 😄

**Comment by victorgarciaesgi** [comment]
This one is hard

**Comment by victorgarciaesgi** [comment]
Found the problem, it was coming from `applyIf`

**Comment by martinszeltins** [comment]
> Fixed in `0.2.11`

Thanks, it works now! 🙏🏻
